src/Tests/Microsoft.NET.Publish.Tests/RuntimeIdentifiersTests.cs

```diff
@@ -194,6 +195,81 @@ public void PublishWithRuntimeIdentifier(bool publishNoBuild)
             }
         }
 
+        [WindowsOnlyTheory]
+        [InlineData(false, false)] // publish rid overrides rid in project file if publishing
+        [InlineData(true, false)] // publish rid doesnt override global rid
+        [InlineData(true, true)] // publish rid doesnt override global rid, even if global
+        public void PublishRuntimeIdentifierSetsRuntimeIdentifierAndDoesOrDoesntOverrideRID(bool runtimeIdentifierIsGlobal, bool publishRuntimeIdentifierIsGlobal)
+        {
+            string tfm = ToolsetInfo.CurrentTargetFramework;
+            string publishRuntimeIdentifier = "win-x64";
+            string runtimeIdentifier = "win-x86";
+
+            var testProject = new TestProject()
+            {
+                IsExe = true,
+                TargetFrameworks = tfm
+            };
+            if (!publishRuntimeIdentifierIsGlobal)
+                testProject.AdditionalProperties["PublishRuntimeIdentifier"] = publishRuntimeIdentifier;
+            if (!runtimeIdentifierIsGlobal)
+                testProject.AdditionalProperties["RuntimeIdentifier"] = runtimeIdentifier;
+            testProject.RecordProperties("RuntimeIdentifier");
+
+            List<string> args = new List<string>
+            {
+                runtimeIdentifierIsGlobal ? $"/p:RuntimeIdentifier={runtimeIdentifier}" : "",
+                publishRuntimeIdentifierIsGlobal ? $"/p:PublishRuntimeIdentifier={publishRuntimeIdentifier}" : ""
+            };
+
+            string identifier = $"PublishRuntimeIdentifierOverrides-{publishRuntimeIdentifierIsGlobal}-{runtimeIdentifierIsGlobal}";
+            var testAsset = _testAssetsManager.CreateTestProject(testProject, identifier: identifier);
+            var publishCommand = new DotnetPublishCommand(Log);
+            publishCommand
+                .WithWorkingDirectory(Path.Combine(testAsset.TestRoot, MethodBase.GetCurrentMethod().Name))
```

Comment:
You should be able to do this instead of getting the method name via reflection:

```suggestion
                .WithWorkingDirectory(Path.Combine(testAsset.TestRoot, testProject.Name))
```